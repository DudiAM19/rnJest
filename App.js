/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React, {useState} from 'react';
import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  useColorScheme,
  View,
  TouchableOpacity,
} from 'react-native';

import {
  Colors,
  DebugInstructions,
  Header,
  LearnMoreLinks,
  ReloadInstructions,
} from 'react-native/Libraries/NewAppScreen';

const App = () => {
  const [darkMode, setDarkMode] = useState(false);

  const changeDarkMode = () => {
    setDarkMode(!darkMode);
  };
  return (
    <View style={styles.newContainer(darkMode)}>
      <StatusBar backgroundColor={darkMode ? '#282C35' : 'white'} barStyle={darkMode ? 'light-content' : 'dark-content'} />
      <TouchableOpacity testID='btnChange' onPress={changeDarkMode} style={styles.button(darkMode)}>
        <Text style={styles.textButton(darkMode)}>Switch</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  newContainer: darkMode => ({
    backgroundColor: darkMode ? '#282C35' : 'white',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  }),
  button: darkMode => ({
    padding: 20,
    backgroundColor: darkMode ? 'white' : '#282C35',
    borderRadius: 20,
  }),
  textButton: darkMode => ({
    color: darkMode ? '#282C35' : 'white',
    fontWeight: 'bold'
  }),
});

export default App;
