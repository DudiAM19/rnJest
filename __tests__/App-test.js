import React from "react";
import { fireEvent, render } from "@testing-library/react-native";
import App from '../App';

describe('testing app', () => {
  it('success', () => {
    const page = render(<App />);
    page.getAllByTestId('btnChange');
  })
})