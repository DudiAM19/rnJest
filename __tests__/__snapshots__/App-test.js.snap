// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#282C35",
        "borderRadius": 20,
        "opacity": 1,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      Switch
    </Text>
  </View>
</View>
`;
